src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -12222,5 +12222,74 @@ static object Execute(Func<object> f)
 0
 {(IntPtr.Size == 4 ? "System.OverflowException" : "0")}");
         }
+
+        [WorkItem(42500, "https://github.com/dotnet/roslyn/issues/42500")]
+        [Fact]
+        public void ExplicitImplementationReturnTypeDifferences()
+        {
+            string source =
+@"struct S<T>
+{
+}
+interface I
+{
+    S<nint> F1();
+    S<System.IntPtr> F2();
+}
+class C : I
+{
+    S<System.IntPtr> I.F1() => default;
+    S<nint> I.F2() => default;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyEmitDiagnostics();
+            var type = comp.GetTypeByMetadataName("I");
+            Assert.Equal("S<nint> I.F1()", type.GetMember("F1").ToTestDisplayString());
+            Assert.Equal("S<System.IntPtr> I.F2()", type.GetMember("F2").ToTestDisplayString());
+            type = comp.GetTypeByMetadataName("C");
+            Assert.Equal("S<System.IntPtr> C.I.F1()", type.GetMember("I.F1").ToTestDisplayString());
+            Assert.Equal("S<nint> C.I.F2()", type.GetMember("I.F2").ToTestDisplayString());
+        }
+
+        [WorkItem(42500, "https://github.com/dotnet/roslyn/issues/42500")]
+        [Fact]
+        public void OverrideParameterCustomModifierDifferences()
+        {
+            var sourceA =
+@".class private System.Runtime.CompilerServices.NativeIntegerAttribute extends [mscorlib]System.Attribute
+{
+  .method public hidebysig specialname rtspecialname instance void .ctor() cil managed { ret }
+}
+.class public A
+{
+  .method public hidebysig specialname rtspecialname instance void .ctor() cil managed { ret }
+  .method public virtual void F1(native int modopt(int32) i)
+  {
+    .param [1]
+    .custom instance void System.Runtime.CompilerServices.NativeIntegerAttribute::.ctor() = ( 01 00 00 00 ) 
+    ret
+  }
+  .method public virtual void F2(native int modopt(int32) i)
+  {
+    ret
+  }
+}";
+            var refA = CompileIL(sourceA);
+
+            var sourceB =
+@"class B : A
+{
+    public override void F1(System.IntPtr i) { }
+    public override void F2(nint i) { }
+}";
```

Comment:
Could we complete the four permutations : base has modified IntPtr vs. nint, and override has unmodified IntPt vs. nint?
Could we also add one nested case?

Reply:
Added all four permutations. Nested cases are covered in the preceding test.

---
In reply to: [485234664](https://github.com/dotnet/roslyn/pull/47536#discussion_r485234664) [](ancestors = 485234664)

Reply:
The problem with the preceding case is it doesn't have modifiers to copy. Are we even entering the `CopyTypeCustomModifiers` method?

---
In reply to: [485239131](https://github.com/dotnet/roslyn/pull/47536#discussion_r485239131) [](ancestors = 485239131,485234664)

Reply:
`CopyTypeCustomModifiers()` is called for overrides and explicit implementations, even without custom modifiers. I've changed the last test here to cover nested cases as well.

---
In reply to: [485242939](https://github.com/dotnet/roslyn/pull/47536#discussion_r485242939) [](ancestors = 485242939,485239131,485234664)

Reply:
Thanks

---
In reply to: [485245803](https://github.com/dotnet/roslyn/pull/47536#discussion_r485245803) [](ancestors = 485245803,485242939,485239131,485234664)